vello/src/scene.rs

```diff
@@ -106,6 +106,37 @@ impl Scene {
         self.encoding.encode_end_clip();
     }
 
+    /// Draw a rounded rectangle blurred with a gaussian filter.
+    pub fn draw_blurred_rounded_rect(
+        &mut self,
+        transform: Affine,
+        brush: Color,
+        size: impl Into<Size>,
+        radius: f64,
```

Comment:
Not because I expect you to do this, but because it's important that it's raised, but is it at all plausible for the corner radii to be different? I.e. make this function accept a [`RoundedRect`](https://docs.rs/kurbo/latest/kurbo/struct.RoundedRect.html) - which despite the docs, can have different radii on each corner since https://github.com/linebender/kurbo/pull/166

I don't have a good understanding of the underlying maths to make this work

Reply:
I thought about this as well but seems tricky at least.
If it would be dependent on the (continuous) distance alone it would be fine in my opinion but the parameters (e.g squircle shape) derived from the corner radius would introduce dicontinuities at the quadrant boundaries, probably visible at larger filter lengths. We could derive shared parameters for the whole shape with some tradeoff in quality but without larger adjustments to the approach.

Reply:
Yeah - that's fine. I've not thought about it deeply, and I don't think that it is a necessary feature.

Thought it would be important to at least ensure that it is discussed, as much for when Raph comes through to gut-check this.

Reply:
> I don't think that it is a necessary feature.

FWIW, this definitely is a necessary feature for CSS box-shadow support. Not sure how common they are in the wild, but the spec certainly allows authors to specify such shadows.

Reply:
https://github.com/DioxusLabs/blitz/blob/main/packages/blitz/src/renderer/multicolor_rounded_rect.rs

we implemented the maths here, you can just copy it if you need to

Reply:
Just rounded rect can be done without too much difficulty in a distance field context by, basically, setting the radius by quadrant. But in the blurred case you will definitely see discontinuities depending on the exact parameters. I think it is possible to fix, but will require some doing.

Of course another possibility is to render the rounded rect as a vector shape (as in the code Jon linked) and blur it, but (a) that's going to be slower than doing it in a shader, even with heavier math, and (b) we don't have the infrastructure in place. It would be nice to have though, as you do need that for text shadows.

I don't recommend using clips for this, it adds complexity and possible performance issues, as well as potential conflation artifacts.

Reply:
I quickly tested how the artifacts look like and there are indeed visible discontinuities at larger kernel sizes (see figure below). It seems to be primarily caused by the `scale` factor. Using a fixed or interpolated value would be an option here. But I would postpone it for now as followup.
![grafik](https://github.com/user-attachments/assets/76af5f2c-713a-434d-8f48-cd045ec74504)